torch/csrc/jit/script/compiler.cpp

```diff
@@ -111,6 +111,22 @@ struct Environment {
 
     return sv;
   }
+
+  SugaredValuePtr createCapturedInputIfNeeded(const SourceRange& loc, std::string ident) {
+    auto in_frame = findInThisFrame(ident);
+    if (in_frame)
+      return in_frame;
+
+    // recursively handles the case where parent blocks are also loops
+    auto from_parent = next ? next->createCapturedInputIfNeeded(loc, ident) : nullptr;
+
+    // recursively create the captured input if it is the loop block
+    if (from_parent && getBlockOwningKind() == prim::Loop) {
+      from_parent = createCapturedInput(from_parent->asValue(loc, method), ident);
```

Comment:
The asValue here should be replaced with:

```
if(Value* simple_val = asSimple(from_parent))
  from_parent = createCapturedInput(simple_val, ident);
```

Like it is in the original getSugaredVar code. Otherwise things like `self` won't be usable inside loops.